diff -uprN distcc-3.3.3/src/prefork.c distcc-3.3.3.new/src/prefork.c
--- distcc-3.3.3/src/prefork.c	2019-08-15 04:08:47.000000000 +0900
+++ distcc-3.3.3.new/src/prefork.c	2020-11-10 20:51:23.736229213 +0900
@@ -57,6 +57,8 @@
 #include "netutil.h"
 #include "stats.h"
 
+extern const char *stats_text[20];
+
 void dcc_manage_kids(int listen_fd);
 static void dcc_sigchld_handler(int sig);
 static void dcc_create_kids(int listen_fd);
diff -uprN distcc-3.3.3/src/serve.c distcc-3.3.3.new/src/serve.c
--- distcc-3.3.3/src/serve.c	2019-08-15 04:08:47.000000000 +0900
+++ distcc-3.3.3.new/src/serve.c	2020-11-10 20:51:31.613051669 +0900
@@ -641,6 +643,7 @@ static int make_temp_dir_and_chdir_for_c
 /**
  * Read a request, run the compiler, and send a response.
  **/
+extern const char *stats_text[20];
 static int dcc_run_job(int in_fd,
                        int out_fd)
 {
diff -uprN distcc-3.3.3/src/stats.h distcc-3.3.3.new/src/stats.h
--- distcc-3.3.3/src/stats.h	2019-08-15 04:08:47.000000000 +0900
+++ distcc-3.3.3.new/src/stats.h	2020-11-10 20:49:41.884211808 +0900
@@ -33,8 +33,6 @@ enum stats_e { STATS_TCP_ACCEPT, STATS_R
                 STATS_COMPILE_OK, STATS_COMPILE_ERROR, STATS_COMPILE_TIMEOUT,
                 STATS_CLI_DISCONN, STATS_OTHER, STATS_ENUM_MAX };
 
-const char *stats_text[20];
-
 int  dcc_stats_init(void);
 void dcc_stats_init_kid(void);
 int  dcc_stats_server(int listen_fd);
