diff -uprN Qengineering-caffe.old/cmake/Cuda.cmake Qengineering-caffe/cmake/Cuda.cmake
--- Qengineering-caffe.old/cmake/Cuda.cmake	2021-06-03 00:47:28.940470209 +0900
+++ Qengineering-caffe/cmake/Cuda.cmake	2021-06-02 07:17:34.895452992 +0900
@@ -190,7 +190,7 @@ function(detect_cuDNN)
     set(HAVE_CUDNN  TRUE PARENT_SCOPE)
     set(CUDNN_FOUND TRUE PARENT_SCOPE)
 
-    file(READ ${CUDNN_INCLUDE}/cudnn.h CUDNN_VERSION_FILE_CONTENTS)
+    file(READ ${CUDNN_INCLUDE}/cudnn_version.h CUDNN_VERSION_FILE_CONTENTS)
 
     # cuDNN v3 and beyond
     string(REGEX MATCH "define CUDNN_MAJOR * +([0-9]+)"
